src/Formats/MarkInCompressedFile.cpp

```diff
@@ -97,7 +97,7 @@ std::tuple<const MarksInCompressedFile::BlockInfo *, size_t> MarksInCompressedFi
 
 size_t MarksInCompressedFile::approximateMemoryUsage() const
 {
-    return sizeof(*this) + blocks.size() * sizeof(blocks[0]) + packed.size() * sizeof(packed[0]);
+    return sizeof(*this) + blocks.allocated_bytes() + packed.allocated_bytes();
```

Comment:
another idea is to use jemalloc's `nallocx` to get exact bin size for all these allocations but maybe it's not needed

E.g. `MarksInCompressedFile` is 56 bytes which falls into 64 byte bin so there is 8 byte difference for each object that we allocate. 
That is not a lot but maybe the difference is bigger for larger allocations like blocks and packed.